packages/core/src/modules/credentials/CredentialsModule.ts

```diff
@@ -354,6 +355,14 @@ export class CredentialsModule<
     const service = this.getService(credentialRecord.protocolVersion)
     await service.updateState(credentialRecord, CredentialState.Declined)
 
+    //  if credential was offered by established connection, attempt to send problem report upon credential decline
+    if (credentialRecord.connectionId) {
+      this.sendProblemReport({
+        message: `Credential Offer ${credentialRecordId} has been declined`,
```

Comment:
The credentialRecordId is irrelevant to the other party, maybe we should use the thread id instead?